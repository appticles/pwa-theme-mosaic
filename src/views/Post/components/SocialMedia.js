import React from 'react';
import PropTypes from 'prop-types';
import { Icon } from 'semantic-ui-react';
import config from 'react-global-configuration';
import './style.css';

const SocialMedia = (props) => {
  // load social media app configuration
  const socialMedia = config.get('socialMedia');

  return (
    <div className="social-media">
      {socialMedia.facebook ? (
        <a href={'https://m.facebook.com/sharer.php?u=' + props.link} target="_blank">
          <Icon name="facebook f" size="large" circular inverted color="blue" />
        </a>
      ) : null}
      {socialMedia.twitter ? (
        <a href={'https://twitter.com/intent/tweet?text=' + encodeURIComponent(props.title) + ' ' + props.link} target="_blank">
          <Icon name="twitter" size="large" circular inverted color="teal" />
        </a>
      ) : null}
      {socialMedia.google ? (
        <a href={'https://plus.google.com/share?url=' + props.link} target="_blank">
          <Icon name="google plus" size="large" circular inverted color="red" />
        </a>
      ) : null}
    </div>
  );
};

SocialMedia.propTypes = {
  link: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};
export default SocialMedia;
