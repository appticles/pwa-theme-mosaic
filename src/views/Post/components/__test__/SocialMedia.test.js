import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import SocialMedia from '../SocialMedia';

describe('Social media component', () => {
  const title = 'mock-title';
  const link = 'mock-link';

  it('should render correctly', () => {
    const output = shallow(<SocialMedia title={title} link={link} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
