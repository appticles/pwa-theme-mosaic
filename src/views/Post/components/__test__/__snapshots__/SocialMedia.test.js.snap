// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social media component should render correctly 1`] = `
<div
  className="social-media"
/>
`;
